sockets/tcp_socket_test.go

```diff
@@ -0,0 +1,45 @@
+package sockets
+
+import (
+	"testing"
+	"crypto/tls"
+
+	"github.com/docker/go-connections/tlsconfig"
+)
+
+const (
+	key = "../tlsconfig/fixtures/key.pem"
+	cert = "../tlsconfig/fixtures/cert.pem"
+)
+
+func TestNewTCPSocket(t *testing.T) {
+	addr := "localhost:1234"
+	versions := []uint16{
+		tls.VersionTLS11,
+		tls.VersionTLS12,
+	}
+
+	for _, v := range versions {
+		tlsConfig, err := tlsconfig.Server(tlsconfig.Options{
+			MinVersion: v,
+			CertFile:   cert,
+			KeyFile:    key,
+		})
+
+		if err != nil {
+			t.Fatal("Unable to configure server TLS", err)
+		}
+
+		if tlsConfig.MinVersion != v {
+			t.Fatal("Unexpected minimum TLS version: ", tlsConfig.MinVersion)
+		}
```

Comment:
I don't think we should be putting any assertion on `tlsconfig` as its not a part of this package